core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/LazyHttpClient.java

```diff
@@ -64,56 +47,43 @@
     @GuardedBy("lock")
     private RuntimeException initException;
     @GuardedBy("lock")
-    private CloseableHttpClient httpClient;
-
-    static ApacheSender43 create() {
-        return new ApacheSender43();
-    }
-
-    private ApacheSender43() {
-    }
+    private CloseableHttpClient delegate;
 
-    @Override
-    public HttpResponse sendRequest(HttpUriRequest request) throws IOException, FriendlyException {
-        try {
-            return getHttpClient().execute(request);
-        } catch (SSLHandshakeException e) {
-            String completeUrl = "https://" + request.getURI().getHost();
-            throw SSLUtil.newSSLFriendlyException(completeUrl);
-        }
+    public static CloseableHttpClient getInstance() {
+        return INSTANCE;
     }
 
-    @Override
-    public void dispose(HttpResponse response) {
-        try {
-            if (response != null) {
-                ((CloseableHttpResponse) response).close();
-            }
-        } catch (IOException e) {
-            logger.error("Failed to send or failed to close response, exception: {}", e.toString());
-        }
+    private LazyHttpClient() {
     }
 
-    @Override
-    public HttpClient getHttpClient() {
+    private CloseableHttpClient getDelegate() {
         synchronized (lock) {
-            if (httpClient != null) {
-                return httpClient;
+            if (delegate != null) {
+                return delegate;
             }
             if (initException != null) {
                 throw initException;
             }
             try {
-                httpClient = init();
+                delegate = init();
             } catch (RuntimeException e) {
                 initException = e;
             }
-            return httpClient;
+            return delegate;
         }
     }
 
-    @Override
-    public void enhanceRequest(HttpPost request) {
+    public static void dispose(HttpResponse response) {
+        try {
+            if (response != null) {
+                ((CloseableHttpResponse) response).close();
+            }
+        } catch (IOException e) {
+            logger.error("Failed to send or failed to close response, exception: {}", e.toString());
+        }
+    }
+
+    public static void enhanceRequest(HttpPost request) {
```

Comment:
does it need to been static?

Reply:
static makes it clear that the method doesn't rely on instance state

Reply:
is it a good practice in java to use a static method whenever it doesn't rely on the instance state? it's just a bit confusing to use non-static and static interchangeably.  are you more concerning about performance gain in this case? Static method does occupy less space and memory allocation happens once,  binding occurs at compile time or earlier time instead of run time, and etc?

Reply:
there's not a perf difference, it's more readability, whether a method is just a "utility" method (input->output) or if it is really part of the object state/behavior, in this case, probably better to move these static methods to some standalone util class, but it's all going to change soon when we move to Azure SDKs

Reply:
util class makes sense to me :)